dbms/src/Parsers/ParserShowTablesQuery.cpp

```diff
@@ -38,27 +39,36 @@ bool ParserShowTablesQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
     {
         query->databases = true;
     }
-    else if (s_tables.ignore(pos, expected))
+    else
     {
-        if (s_from.ignore(pos, expected))
-        {
-            if (!name_p.parse(pos, database, expected))
-                return false;
-        }
+        if (s_temporary.ignore(pos))
+            query->temporary = true;
 
-        if (s_not.ignore(pos, expected))
-            query->not_like = true;
-
-        if (s_like.ignore(pos, expected))
+        if (s_tables.ignore(pos, expected))
         {
-            if (!like_p.parse(pos, like, expected))
+            if (s_from.ignore(pos, expected))
+            {
+                if (query->temporary)
+                    throw Exception("Unable to parse FROM,Because the temporary table does not have a database.");
```

Comment:
Add an error code.
I would suggest `ErrorCodes::SYNTAX_ERROR`, but I think it is not syntax error and the check should be done in the interpreter.